devtools/maven/src/main/java/io/quarkus/maven/AbstractDeploymentMojo.java

```diff
@@ -41,16 +43,18 @@ protected void doExecute() throws MojoExecutionException {
     }
 
     public Deployer getDeployer() {
-        return Deployer.getDeployer(mavenProject())
-                .orElse(Deployer.kubernetes);
+        return getDeployer(Deployer.kubernetes);
     }
 
-    public Optional<String> getImageBuilder() {
-        return Optional.ofNullable(imageBuilder);
+    public Deployer getDeployer(Deployer defaultDeployer) {
+        return Deployer.getDeployer(mavenProject())
+                .orElse(defaultDeployer);
     }
 
     @Override
     protected List<Dependency> forcedDependencies(LaunchMode mode) {
+        if (!forceDependencies)
```

Comment:
If I get it right, you are addding this, so that we don't go through the maven project dependencies if a target is not specified right?
I think that a comment here would help.